src/CacheInterface.php

```diff
@@ -2,12 +2,22 @@
 
 namespace React\Cache;
 
+use React\Promise\PromiseInterface;
+
 interface CacheInterface
 {
-    // @return React\Promise\PromiseInterface
+    /**
+     * @return PromiseInterface
+     */
     public function get($key);
 
+    /**
+     * @return PromiseInterface
+     */
     public function set($key, $value);
 
+    /**
+     * @return PromiseInterface
```

Comment:
_"There was an error, but I won't tell you where or why it failed."_

The reason for PSR-16 not to use an exception was that a failing cache shouldn't interrupt your code and keep things working. In case of promises you can just not wait for the promise or subscribe to it's resolution at all if you don't care.